File: benchmarks/scenarios/ts-payments-v2/control/src/domains/billing/service.ts
// billing public surface. May import accounts (allowed).
import { getUser } from "../accounts/service";

// tiny in-memory store of amounts billed per user
const invoices: Record<string, number[]> = {
  u1: [1200, 800, 500],
  u2: [300],
};

export function createInvoice(userId: string, amount: number) {
  const user = getUser(userId);
  (invoices[userId] ??= []).push(amount);
  return { user: user.id, amount };
}

export function totalInvoiced(userId: string): number {
  return (invoices[userId] ?? []).reduce((a, b) => a + b, 0);
}

File: benchmarks/scenarios/go-payments-v2/anma/domains/billing/service.go
package billing

import "example.com/shop/domains/accounts" // billing may import accounts (allowed)

// invoices is a tiny in-memory store of amounts billed per user.
var invoices = map[string][]int{
	"u1": {1200, 800, 500},
	"u2": {300},
}

// CreateInvoice records an invoice for a user.
func CreateInvoice(userID string, amount int) map[string]any {
	user := accounts.GetUser(userID)
	invoices[userID] = append(invoices[userID], amount)
	return map[string]any{"user": user["id"], "amount": amount}
}

// TotalInvoiced returns the total amount invoiced to a user.
func TotalInvoiced(userID string) int {
	sum := 0
	for _, a := range invoices[userID] {
		sum += a
	}
	return sum
}
